.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -2044,4 +2044,28 @@ stages:
           parameters:
             build: true
             baseImage: debian
-            command: "CheckBuildLogsForErrors"
\ No newline at end of file
+            command: "CheckBuildLogsForErrors"
+
+- stage: trace_pipeline
+  condition: eq(variables['isScheduledBuild'], 'False')
+  dependsOn: [integration_tests_windows, integration_tests_windows_iis, integration_tests_linux, unit_tests_linux, unit_tests_macos, unit_tests_windows, unit_tests_arm64, master_commit_id, integration_tests_arm64, benchmarks, throughput, throughput_appsec, tool_artifacts_tests_windows, tool_artifacts_tests_linux, upload_to_s3, upload_to_azure, installer_smoke_tests, installer_smoke_tests_arm64]
```

Comment:
yuk, but what choice doo we have 😂 

Although, what happens if one of those _doesn't_ run 🤔 e.g. early build fails, so one of them never runs. AFAIK, this will never be triggered? So we still don't get monitoring if a build fails 🤔

Reply:
Though, I didn't try to kill a n-2 stage. Testing now. I thought it would work because I didn't add the `succeeded() `funciton in the condition. But better to test it indeed

Reply:
So the second test works as well https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=98128&view=results. I just need to rerun my pipeline now :)

Reply:
Excellent, looking good 👍